src/types/models/token-intents.ts

```diff
@@ -16,20 +17,32 @@ interface TokenIntentCardDetails {
   };
 }
 
-interface TokenIntent<DataType = DataObject>
-  extends Omit<Auditable, 'modifiedAt' | 'modifiedBy'> {
-  data: TokenData<DataType>;
-  type: TokenType;
-  enrichments?: TokenIntentCardDetails;
-  tenantId: string;
-  expiresAt: string;
-}
+type TokenTypesForTokenIntents = Exclude<TokenType, 'token' | 'card'>;
+
+type TokenTypeMap = {
+  [K in TokenTypesForTokenIntents]: {
+    type: K;
+  } & Record<K, Record<string, unknown>>;
+};
+
+type TokenIntent<DataType = DataObject> = (TokenBase<DataType> &
+  Omit<Auditable, 'modifiedAt' | 'modifiedBy'> & {
+    id: string;
+    tenantId: string;
+    expiresAt: string;
+    fingerprint?: string;
+  }) &
+  (
+    | TokenTypeMap[TokenTypesForTokenIntents]
+    | TokenIntentCardDetails
+    | {
+        type: 'token';
+      }
+  );
```

Comment:
confusing types here. this essentially defines a property on `TokenIntents` that looks like a token type that gets passed in. so this would be a possible token-intent:
```
const tokentIntent: TokenIntent = {
  ...
  type: 'bank',
  bank: { ... }
}
```
I've got the `card` prop that comes back defined in the `TokentIntentCardDetails` type. and the `token` type will not return a `token` prop